content/desktop/release-notes.md

```diff
@@ -21,11 +21,73 @@ Docker Desktop versions older than 6 months from the latest release are not avai
 
 Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/projects/1) to see what's coming next.
 
+## 4.31.0
+
+{{< release-date date="2024-05-30" >}}
+
+{{< desktop-install all=true beta_win_arm=true version="4.31.0" build_path="/TODO_NOT_AVAILABLE_YET/" >}}
+
+### New
+
+- Admins can now use alternative methods to the `registry.json` file, to enforce sign in.
+- [Air-Gapped Containers](desktop/hardened-desktop/air-gapped-containers.md) is now generally available. 
+- Added Multi-node Kind as a Kubernetes option (Beta).
+
+#### For all platforms
+
+- Introducing: compose bridge. You might be asked your password to install it. Make sure in settings->advanced->Automatically check configuration is turned on.
+- Added a Docker Compose documentation helper and a new way to view Compose files with the Compose File Viewer (Beta).
+
+### Upgrades
+
+- [Docker Engine and CLI v26.1.3](https://github.com/moby/moby/releases/tag/v26.1.3).
+- [Docker Scout CLI v1.9.1](https://github.com/docker/scout-cli/releases/tag/v1.9.1)
+- [Docker Compose v2.27.1](https://github.com/docker/compose/releases/tag/v2.27.1)
+- [Docker Buildx v0.14.1](https://github.com/docker/buildx/releases/tag/v0.14.1)
+- [Containerd v1.6.32](https://github.com/containerd/containerd/releases/tag/v1.6.32)
+- [Credential Helpers v0.8.2](https://github.com/docker/docker-credential-helpers/releases/tag/v0.8.2)
+- Linux kernel `v6.6.31`
+
+### Bug fixes and enhancements
+
+#### For all platforms
+
+- Allow newer release to be displayed when an update has been already downloaded.
+- Moved the setting **Automatically check configuration** from **Advanced** settings to **General** settings.
+- Fixed a bug where Docker Debug was not working properly with Enhanced Container Isolation enabled.
+- Fixed a bug where UDP responses were not truncated properly.
+- Fixed a bug where the Update screen was  hidden when using [Settings Management](hardened-desktop/settings-management/_index.md).
+- Fixes [docker/for-mac#7274](https://github.com/docker/for-mac/issues/7274)
+
+#### For Windows
+
+- Changed the `--allowed-org` installer flag to write a policy registry key instead of to the `registry.json`.
+
+#### For Mac
+
+- Improved VirtioFS caching by implementing longer attributes timeout and invalidation.
+
+#### For Linux
+
+- Added Linux headers to the VM, to ease the compilation of custom kernel modules.
+
+### Security
+
+#### For all platforms
+
+- Fixed a security bug in Enhanced Container Isolation (ECI) mode where a user could create Docker volumes sourced from restricted directories inside the Docker Desktop VM and mount them into containers, thereby giving the container access to such restricted VM directories.
+
+### Deprecation
```

Comment:
cc @laurazard @krissetto - we should probably mention the removal of the "cloud integration" wrapper (and the "com.docker.cli" to be removed in the next release)

Reply:
```suggestion
### Deprecation
- The CLI binary that used to be shipped as `com.docker.cli` is now shipped simply as `docker`. This release leaves the CLI binary as `com.docker.cli`, but it will be removed next release.
```